scripts/scr_battle_roster/scr_battle_roster.gml

```diff
@@ -42,32 +42,40 @@ function scr_battle_roster(_battle_loci, _loci_specific, _is_planet) {
 
 
 	// if (_is_planet=true){
-	    var company, v, meat;
+	    var company, v, meat, marine;
 	    company=-1;v=0;meat=false;
+		deploying_unit=obj_ini;
 
 	    instance_activate_object(obj_pnunit);
 
+		//For each company and the HQ
 	    repeat(11){
 			if (man_limit_reached == true){break;}
 	        company+=1;v=0;okay=0;
-	        repeat(300){
+			//For each marine in that company, while unit exists (either marine name or vehicle role, vehicles have no names saved)
+			//Marines and vehicles get added AT THE SAME TIME, (index [0][1] adds marine AND vehicle at index at the same time for loop x)
+			//This is possible since array for saving vehicles and marines are separated
+			//v<300 based on previous repeat(300), so it might be nonsense
+			while ((deploying_unit.name[company,v]!=""      || 
+					deploying_unit.veh_role[company,v]!="")    && v<300){
 				if (man_limit_reached == true){break;}
+				//array[0] set to 0, so the proper array starts at array[1], for some reason
 				v+=1;
 
 	            if (stop=0){
 	                if (!instance_exists(obj_drop_select)){// Only when attacked
-	                    if (_is_planet=true) and (obj_ini.loc[company,v]=fiy) and (obj_ini.wid[company,v]=secc) and (obj_ini.hp[company,v]>0) and (obj_ini.god[company,v]<10) then okay=1;
-	                    if (_is_planet=false) and (obj_ini.lid[company,v]=_loci_specific) and (_loci_specific=_loci_specific) and (obj_ini.hp[company,v]>0) and (obj_ini.god[company,v]<10) then okay=1;
+	                    if (_is_planet=true) and (deploying_unit.loc[company,v]=fiy) and (deploying_unit.wid[company,v]=secc) and (deploying_unit.hp[company,v]>0) and (deploying_unit.god[company,v]<10) then okay=1;
+	                    if (_is_planet=false) and (deploying_unit.lid[company,v]=_loci_specific) and (deploying_unit.hp[company,v]>0) and (deploying_unit.god[company,v]<10) then okay=1;
 	                }
 	                if (instance_exists(obj_drop_select)){
 	                    if (obj_drop_select.attack=1){
-	                        if (_is_planet=true) and (obj_ini.loc[company,v]=fiy) and (obj_ini.wid[company,v]=secc) and (obj_ini.hp[company,v]>0) and (obj_ini.god[company,v]<10) then okay=1;
-	                        if (_is_planet=false) and (obj_ini.lid[company,v]=_loci_specific) and (_loci_specific=_loci_specific) and (obj_ini.hp[company,v]>0) and (obj_ini.god[company,v]<10) then okay=1;
+	                        if (_is_planet=true) and (deploying_unit.loc[company,v]=fiy) and (deploying_unit.wid[company,v]=secc) and (deploying_unit.hp[company,v]>0) and (deploying_unit.god[company,v]<10) then okay=1;
+	                        if (_is_planet=false) and (deploying_unit.lid[company,v]=_loci_specific) and (deploying_unit.hp[company,v]>0) and (deploying_unit.god[company,v]<10) then okay=1;
```

Comment:
why 10? can we replace it with _company_amount or something?

Reply:
Not sure what you are refering to. The obj.init **god** atribute? Not so sure yet of what it does if that's the case.

Reply:
I meant this part:

(deploying_unit.god[company,v]<10)

this looks like a check if company number is below 10 (the max company limit)

Would be better to replace the literal 10 with a (static?) variable _max_company_limit = 10, or something like this

Reply:
I believe that variable might be related to the chaos/corruption mechanic. I believe it's not even implemented, but we got some code that might have been intended to be part some sort of internal chaos vs loyalist struggle inside the chapter, which would lead to a battle, and would explain why we that check on scr_battle_roster (god>=10 would mean a traitorous/corrupted Astartes, if I'm right, so the marine would need to be excuded from the player battle roster).

I do not believe it's implemented though, since the atribute it's always set to 0 for every marine, so its currently irrelevant.

The company number of the marine/unit it's saved in the 2D Array structure itself, i believe (array_atribute[company_number][marine_number_in_the_company]).